src/common/reducers/entities/index.js

```diff
@@ -1,27 +1,43 @@
 import merge from 'lodash/merge';
 
 import * as RepoActionTypes from '../../actions/repository';
-
 import repository from './repository';
 
-
 // TODO snaps
 
-export function entities(state = { snaps: {}, repos: {} }, action) {
+export function entities(state = {
+  snaps: {},
+  repos: {},
+  owners: {}
+}, action) {
   if (action.payload && action.payload.entities) {
     return merge({}, state, action.payload.entities);
   }
 
   // only modify repos if action is one of REPO_ types
-  if (action.payload && action.payload.id && RepoActionTypes[action.type]) {
+  if (RepoActionTypes[action.type]) {
+    return reduceRepoEntity(state, action);
+  }
+
+  return state;
+}
+
+function reduceRepoEntity(state, action) {
+  return reduceEntityHelper(state, action, repository, 'repos');
+}
+
+function reduceEntityHelper(state, action, reducer, name) {
```

Comment:
This could maybe be made more magical if name were also the key to an object containing reducers.